pyccel/parser/semantic.py

```diff
@@ -2289,6 +2289,7 @@ def _visit_CodeBlock(self, expr):
                 if len(line) != 1:
                     errors.report(f"Variable {line[0]} cannot have multiple types",
                             severity='error', symbol=line[0])
+                ls.extend(line)
```

Comment:
Please explain this change. I don't think it is necessary and am worried that it might break something

Reply:
In the case of `int : list[int]`, the `args` holds `int`, so we need to check for it.

Reply:
Without this line, the `codeBlock` is returns  empty object because `ls` is still empty if it enters this condition.

Reply:
Hmm, this seems like a problem with the printing in Python not a general translation problem

> because ls is still empty if it enters this condition.

I disagree that this is necessarily the case

Reply:
if we tried to translate this code :
```python
def func():
    a : int
```
python translation :
```python
def func():
    pass
```
The only reason I found for this is that the `Codeblock` object is empty

Reply:
> python translation :

Is this the translation on this branch or the devel branch? If it is broken on the devel branch then you should open an issue about that

Reply:
> I agree. What is your point?

In the case of `a: int`, the `ls` variable doesn't take the value of `line`. This is why I added this line, to prevent the loss of the value of line and avoid returning an empty object.

Reply:
> to prevent the loss of the value of line

The information is not lost it is saved elsewhere.

You do indeed arrive here if your code only contains `a : int` but that is not the only case and this doesn't guarantee that `ls` is empty.

> and avoid returning an empty object.

Why does this need to be avoided in all languages? Specifically can you explain why this should be avoided for Fortran and C (as these are the languages for which the translation needs to be optimised)

Reply:
> > python translation :
> 
> Is this the translation on this branch or the devel branch? If it is broken on the devel branch then you should open an issue about that
 
The devel branch.

Reply:
> The devel branch.

The devel branch generates:
```python
def func():
    pass
```
?
Then I don't understand what is the problem here

Reply:
Fixed : 01150c5